File: Daily/PY/LCP7-传递信息.py
from typing import List

class Solution:
    def numWays(self, n: int, relation: List[List[int]], k: int) -> int:
        edges = [[] for _ in range(n)]
        for s2d in relation:
            edges[s2d[0]].append(s2d[1])
        ans = [0]
        def dfs(idx,step,ans):
            if step == k:
                if idx == n - 1:
                    ans[0] += 1
                return 
            for dst in edges[idx]:
                dfs(dst,step + 1,ans)
        dfs(0,0,ans)
        return ans[0]

File: Daily/sbgo/LCP7-传递信息.go
package sbgo

func numWays(n int, relation [][]int, k int) int {
	edges := make([][]int, n)
	for _, s2d := range relation {
		edges[s2d[0]] = append(edges[s2d[0]], s2d[1])
	}
	var dfs func(int, int)
	ans := 0
	dfs = func(idx, step int) {
		if step == k {
			if idx == n-1 {
				ans += 1
			}
			return
		}
		for _, dst := range edges[idx] {
			dfs(dst, step+1)
		}
	}
	dfs(0, 0)
	return ans
}
